plugin/core/registry.py

```diff
@@ -144,11 +143,32 @@ def client_with_capability(self, capability: str) -> Optional[Client]:
         return client_from_session(session_for_view(self.view, capability))
 
 
-class LspRestartClientCommand(sublime_plugin.TextCommand):
+class LspRestartAllServersCommand(sublime_plugin.TextCommand):
     def is_enabled(self) -> bool:
         return is_supported_view(self.view)
 
     def run(self, edit: Any) -> None:
         window = self.view.window()
         if window:
             windows.lookup(window).restart_sessions()
+
+
+class LspRestartServerCommand(sublime_plugin.TextCommand):
+    def is_enabled(self) -> bool:
+        window = self.view.window()
+        if not window:
+            return False
+        return is_supported_view(self.view) and windows.lookup(window).has_active_sessions()
+
+    def run(self, edit: Any) -> None:
+        self.window = self.view.window()
+        if not self.window:
+            return
+
+        self.lsp_sessions = windows.lookup(self.window).get_sessions()
+        if self.lsp_sessions:
+            self.window.show_quick_panel(self.lsp_sessions, self.restart_server)
```

Comment:
Not a good practice to store stuff on the instance as it's per-view, it's not per command execution. You can create lambda:

```suggestion
        sessions = windows.lookup(self.window).get_sessions()
        if sessions:
            self.window.show_quick_panel(self.lsp_sessions, lambda index: self.restart_server(sessions[index]))
```
And update `restart_server` accordingly.

Although you'll get -1 for index when canceling quick panel so you will have to do a bit of error checking. Maybe pass `index` and `sessions` to `restart_servers`.